sklearn/metrics/scorer.py

```diff
@@ -33,14 +34,15 @@
 
 
 class _BaseScorer(six.with_metaclass(ABCMeta, object)):
-    def __init__(self, score_func, sign, kwargs):
+    def __init__(self, score_func, sign, kwargs, deprecation_msg=None):
         self._kwargs = kwargs
         self._score_func = score_func
         self._sign = sign
+        self._deprecation_msg = deprecation_msg
 
-    @abstractmethod
     def __call__(self, estimator, X, y, sample_weight=None):
-        pass
+        if self._deprecation_msg is not None:
+            warnings.warn(self._deprecation_msg, category=DeprecationWarning)
```

Comment:
Don't you want a stacklevel=2 at least here, maybe even 3 (given that this method will always be called inside another)